library/core/src/cell.rs

```diff
@@ -1698,6 +1694,29 @@ impl<T: ?Sized> UnsafeCell<T> {
         // no guarantee for user code that this will work in future versions of the compiler!
         this as *const T as *mut T
     }
+
+    /// Returns a mutable reference to the underlying data.
+    ///
+    /// This call borrows the `UnsafeCell` mutably (at compile-time) which
+    /// guarantees that we possess the only reference.
```

Comment:
In the forum discussion, this was also related to explaining better that `UnsafeCell` is not magic wrt `&mut`. Maybe that is worth emphasizing specifically in the type docs, and then link to this method there?

Reply:
Done:

  - [rendered documentation](https://modest-dubinsky-1f9f47.netlify.app/core/cell/struct.unsafecell)